ase/api/src/wsid_list.c

```diff
@@ -209,31 +209,3 @@ struct wsid_map *wsid_find(struct wsid_tracker *root, uint64_t wsid)
 	return tmp;
 }
 
-/**
- * @ brief Find entry in linked list
- *
- * @param root
- * @param index
- *
- * @return
- */
-struct wsid_map *wsid_find_by_index(struct wsid_tracker *root, uint32_t index)
```

Comment:
You're sort of going the opposite direction we want here. I know this function isn't used in ASE, but the point was that the file is as close as possible to to the OPAE source. At one point, I think it _was_ identical. Aren't we trying, ultimately, to share sources with OPAE instead of making ASE an OPAE reimplementation?

Reply:
This function is dead code in ASE and I don't know whether it is used in OPAE. Will check.

ASE is going to be implemented as a plugin of OPAE like xfpga. ASE can have its own internal implementation, but OPAE API will be the same.

Reply:
It's used in OPAE for MMIO.

Wouldn't it be a better goal to share OPAE sources whenever possible instead of just acting as a separate plugin? The fewer opportunities for divergence, the better.

Reply:
Currently it is not used in ASE so we should remove it. If we need it in the future, we can easily add it back. 

ASE will be added as a plugin after I complete the ASE unit tests. So there will be three plugins for OPAE: xfpga, remote fpga and ASE. I think it is good to integrate ASE as part of OPAE since it will share the same libopae-c (FPGA API) with other plugins while it still keeps its own implementation since RTL simulation is totally different from the actual FPGA hardware implementation.